### PR TITLE
Create env for testing and initial tests

### LINKED ISSUE #190: Test checking that sendInitPing is called on every init of extension
Function `pp-ga.core.sendInitPing` sends ping to server. We sync GA cookies between pseudo-`chrom-extension://hashgoijgsdij...` domain and real `przypispowszechny.pl` domain that way. 

The most robust way to achieve that sync is to make that ping every time extension starts (that's not often).

### LINKED ISSUE #191: Introduce integration/functional tests
In the beginning it can be any solution - let's see what use of it we can have now and with our needs growing will develop initial solution on change it or drop it.

Initial need is test described in #190

### LINKED ISSUE #193: Introduce unit tests
In the beginning it can be any solution - let's see what use of it we can have now and with our needs growing will develop initial solution on change it or drop it.

Initial need is test described in #192

### LINKED ISSUE #222: Refine e2e setup
- [x] Run e2e tests against 100% production build, but with different services
  - using GA_ID_DEV
  - connecting to localhost on specific port (not necessarily 8000)
- [x] Split unit tests and build&e2e to 2 jobs, so that after build it can be run in parallel
- [x] Add some docs about running tests

### PR DESCRIPTION
**Pull request by glowka**
Resolves #191 
Resolves #193 
Resolves #190 
Resolves #222

### CODE REVIEW

**Comment by glowka** [comment]
@kaglowka I added more verbose validation of PP_API setting :)

**Comment by glowka** [comment]
@kaglowka Shall we merge it or is there anything you want to add to that?

File: README.md [hunk 1]
```diff
@@ -135,51 +142,77 @@ and introduce changes to the code just then, rather than build only when it's ne
...
 ```
-npm run start-extension -- --env.api=local
+which is equivalent to
 ```
-#### Documentation of API
+PP_API=local npm run start
+```
+All development values of `PP_API`:
+ - `local` - localhost port 8000
+ - `local-alt` - localhost port 8080
```
Comment (kaglowka) [inline README.md]: There's a potential clash with localhost client application, but we may change the client application port to a different one, too, when we're working on it soon.

File: README.md [hunk 2]
```diff
@@ -183,6 +194,25 @@ So for example, to see the shape of the backend you are developing to, when runn
...
+```
+
+### e2e Tests
+
+Build and then run e2e tests, for example:
+```
+PP_API=local npm run build && PP_API=local npm run e2e 
+```
+Notes:
+- `PP_API` for **build and e2e have to match**
```
Comment (kaglowka) [inline README.md]: :+1:

File: config/browser-extension/ext.prod.config.js [hunk 3]
```diff
@@ -2,15 +2,12 @@ const merge = require('webpack-merge');
 const CreateFileWebpack = require('create-file-webpack');
 
 const common = require('./ext.base.config');
+const packageConf = require('../../package');
 
 // Constant key and id from our production setting
 const KEY = 'MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEArkmF/06jK2b976qLc1RoSyrhGMN1QkbRCbDLnMGL+RfOlZdbDBwXxtTXcnzm2uwUlrZKCaecuD0GOrnOfihaqQv3cVRbRTCWhvyhswqiWea8YA8P0pjiTrsuth6l0NK9WqKYLTUxP+oM9H0nDaOg5I9Jzh5kCZPpwavbx1naXCv7PPcIoGNRgA4y3YJL8qiJx9eY0I6yRdCSS8JX4AeZ4K1ShnzPg5Wih7NZdJlmWCjOajdF8GHxYGt68eGOamIlQYwFzAjinCN+DbI1f4suQcq5oJzw3NCBmCvepaw5f/ivhzCsw0aRfVqRk6EVxhQvRwTBxeBQQ28j5JecxtfN5wIDAQAB'
```
Comment (kaglowka) [inline config/browser-extension/ext.prod.config.js]: I suppose we could also move the key `packageConf` -- they are very closely related and whenever one changes, so does the other (not that we expect it to happen any time soon, but...)

**Pull request merged**